lib/lwm2m-server.js

```diff
@@ -103,6 +104,29 @@ function validateTypes(config, callback) {
     callback(error);
 }
 
+/**
+ * If Lifetime Resource exists, then the registration SHOULD be removed by the Server if a new registration or update
+ * is not received within this lifetime.
+ */
+function checkLifetime() {
+    var CHECK_INTERVAL = 1000;
```

Comment:
As I understand, the `CHECK_INTERVAL` parameter is not the "default server lifetime", but the "minimum interval between lifetime checks". I think the later is a parameter for the server administrator to decide. Just look at the first instruction the checkLifetime() executes: it's a query for all the devices in the database, and a loop over each one of them to test their lifetimes. If the database contains lots of devices, this operation is going to consume a lot of resources, so the service administrator surely would want to decide the check interval (as a smaller interval will be harder to sustain than a bigger one).

In relation with that, I'm starting to think this function should belong to the  registry module, instead of the main `lwm2m-server` one. The reason is that different storage engines may provide different mechanisms to address the same concern of record expiration. For instance, MongoDB provides TTL for records, that may be used in order to avoid using a explicit loop to check the records individually. This method, though, would be needed for the in-memory registries, that don't provide those features. 

Summarizing, I think this function should be moved to both registries (in-memory and mongodb), with the idea to implement TTL-based expiration in MongoDB in a next step. I can do the MongoDB part in a different PR if you make this refactor first (and I'll also create some tests to cover this functionality in the test base).